src/Compiler/Checking/Expressions/CheckComputationExpressions.fs

```diff
@@ -1900,105 +1900,162 @@ let rec TryTranslateComputationExpression
-            if isNil andBangs then
-                error (Error(FSComp.SR.tcInvalidUseBangBinding (), mBind))
-            else
-                let m =
-                    match andBangs with
-                    | [] -> comp.Range
-                    | h :: _ -> h.Trivia.AndBangKeyword
+                    error (Error(FSComp.SR.tcInvalidUseBangBindingNoAndBangs (), m))
+                else
+                    if ceenv.isQuery then
+                        error (Error(FSComp.SR.tcBindMayNotBeUsedInQueries (), mBind))
+
+                    if
+                        isNil (
+                            TryFindIntrinsicOrExtensionMethInfo
+                                ResultCollectionSettings.AtMostOneResult
+                                cenv
+                                ceenv.env
+                                mBind
+                                ceenv.ad
+                                "Using"
+                                ceenv.builderTy
+                        )
+                    then
+                        error (Error(FSComp.SR.tcRequireBuilderMethod "Using", mBind))
+
+                    if
+                        isNil (
+                            TryFindIntrinsicOrExtensionMethInfo
+                                ResultCollectionSettings.AtMostOneResult
+                                cenv
+                                ceenv.env
+                                mBind
+                                ceenv.ad
+                                "Bind"
+                                ceenv.builderTy
+                        )
+                    then
+                        error (Error(FSComp.SR.tcRequireBuilderMethod "Bind", mBind))
+
+                    let bindExpr =
+                        let consumeExpr =
+                            SynExpr.MatchLambda(
+                                false,
+                                mBind,
+                                [
+                                    SynMatchClause(
+                                        pat,
+                                        None,
+                                        TranslateComputationExpressionNoQueryOps ceenv innerComp,
+                                        innerComp.Range,
+                                        DebugPointAtTarget.Yes,
+                                        SynMatchClauseTrivia.Zero
+                                    )
+                                ],
+                                DebugPointAtBinding.NoneAtInvisible,
+                                mBind
+                            )
+
+                        let rhsExpr =
+                            mkSourceExprConditional isFromSource rhsExpr ceenv.sourceMethInfo ceenv.builderValName
 
-                error (Error(FSComp.SR.tcInvalidUseBangBindingNoAndBangs (), m))
+                        mkSynCall "Bind" mBind [ rhsExpr; consumeExpr ] ceenv.builderValName
+                        |> addBindDebugPoint spBind
```

Comment:
What is the difference between this block and the block at 1964-1980, apart from the different inner expression passed in to the `SynMatchClause` inside? ( = the inner expression could be parametrized)

Reply:
Done